### ISSUE TITLE
Update Add Complainant/Witness/Accused Button for Hawaii

### ISSUE THREAD

**Issue by andrew-rist-tw** [PublicDataWorks/complaint-manager#250]
Blocked by #246 

### Business Context:
For Hawaii we want to get rid of the Menu button concept for adding more people to the case after its creation. Instead we want to get closer to what we have on the CreateCaseDialog (though not exactly like it), primarily we want to add the dropdown with person types to the dialog (currently the CivilianDialog) to choose what type you're adding, then let it show the form in the case of a person type with `createDialogAction` set to `showForm` and otherwise (person in custody) just show a button to redirect to the Person in Custody search page

**As a** Hawaii CSOC employee
**I want** to be able to add people to a case after creating it
**So that** I can adjust to changes and add multiple people including Witnesses and Accused (not just Complainant)

### In Scope
1. Change "+ Add Complainant", "+ Add Witness", and "+ Add Accused" buttons on the case details page so that they immediately launch the Civilians dialog (feel free to rename) rather than opening a menu
2. Add dropdown to the CiviliansDialog so that the person type can be chosen
3. Add logic on the Civilians Dialog so that the form is not shown for person in custody
4. Update the Create Civilian service handler to record the `personType`
5. Make sure that all person types can be added appropriately

### Out of Scope
1.
2.
3.

### Open Questions
1.
2.
3.

Steps:
1. Click on add Witness
2. see the modal pop up (with two things: role on case, and person type) [default role: whichever button was pressed,  default person type: person in custody 
3. once both are selected
4. then proceed as normal for selected person type

## Acceptance Criteria

1. Scenario: Click the Add X button
Given I am on the case details page
When I select add Complainant/Witness/Accused
Then I see the popup (in mocks)

2. Scenario: Select person type
Given I am on the add person popup
AND the button i clicked (C/W/A) has been prepopulated
When I select the person type
Then I see the fields relevant to that person type (see person type field in instance files (create dialog action))

3. Scenario: Click Create
Given I have filled out required info for a person
When I select Create
Then the person is added

### Checklist (should be reviewed before grooming as the BA or Pair are writing the card—NOT reviewed for the first time at grooming)
0. Discuss the technical implementation of this card.
1. Discuss how this feature would be included in or affect the Audit Log.
3. In respect to security, what could go wrong? Brainstorm threats using STRIDE model, especially if this card introduces a new technical architecture or design pattern. (TIMEBOX: 7 mins) https://docs.google.com/document/d/18AFl1d3SRieI4d_8R3GyEazqQOycLNDHGN_vOYIHuiE/edit#heading=h.aio9zb5j63wl
7. Ensure any open questions have been resolved with stakeholders.
8. Determine if documentation should be referenced or created.
9. Ensure to include all necessary mockups to this card.
10. Discuss what success/failure user feedback messages would be present, if any occur (front end/back end).
11. Discuss why this feature would not be implemented behind a toggle / what would happen if this card was partially done while a release branch was being cut.
12. If this card is part of a sequence of cards, discuss if this card should be blocked until the testing of previous  cards in the sequence have been finished.
13. Have we added edge cases as ACs that we're considered from a QA perspective
14. Does this card overlap with any entities outlined in [API Housekeeping](https://www.pivotaltracker.com/story/show/173963545)?  

**Notes:**

**Comment by Concepcion-c** [comment]
![Frame 1](https://user-images.githubusercontent.com/121635387/220681049-b9648467-6890-4d7f-a8c9-412f57a358b3.png)
![Frame 2](https://user-images.githubusercontent.com/121635387/220681054-1ab23c73-2799-4b8d-abbe-ef2dc6a09075.png)
![Frame 3](https://user-images.githubusercontent.com/121635387/220681056-d018db10-afab-49a0-b3a9-8f7fe61f1dda.png)
![Frame 4](https://user-images.githubusercontent.com/121635387/220681060-fba85e0e-25c6-4743-adaa-38d26b3f5063.png)
![Frame 5](https://user-images.githubusercontent.com/121635387/220681063-03abe584-85c3-455b-b999-724f1f6bcec4.png)

**Comment by Concepcion-c** [comment]
Or
![Frame 6](https://user-images.githubusercontent.com/121635387/221033477-62415db8-5e4a-43ee-9dc2-209502cfc713.png)
![Frame 7](https://user-images.githubusercontent.com/121635387/221033496-136d791b-a2f6-4a1a-add3-1f4dc43b4aa4.png)

**Comment by Concepcion-c** [comment]
![Frame 6](https://user-images.githubusercontent.com/121635387/221033713-0196ae0c-94f9-4bf2-a363-98833818b45d.png)
![Frame 7](https://user-images.githubusercontent.com/121635387/221033716-04750d0c-2bec-4a37-ab4b-4a8451ced341.png)

**Comment by andrew-rist-tw** [comment]
The PERSON_TYPE object specifies which person type shows a form and which types redirect to a search page. You can reuse that (and use CreateCaseDialog.js as an example for the logic)
